src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -37,6 +37,14 @@ interface UserProvidedValue<T = unknown> {
 
 type UserProvided<T = unknown> = Record<string, UserProvidedValue<T>>;
 
+function assertSavedObjectsClient(
+  savedObjectsClient?: SavedObjectsClientContract
+): asserts savedObjectsClient is SavedObjectsClientContract {
+  if (!savedObjectsClient) {
+    throw new Error('Saved Objects Client is not available.');
+  }
+}
+
 export class UiSettingsClient implements IUiSettingsClient {
```

Comment:
__note:__ not related to this PR, but while reading the code for the UI settings client and how it's used today, I noticed that we don't respect ui settings overrides from config for the anonymous applications (e.g. login page) since they rely on `getRegistered` method. For that reason, for example, login page will always use light theme even if you set dark mode in `yaml` file with `uiSettings.overrides.theme:darkMode: true`.